api/src/main/java/com/community/api/services/ServiceProvider/ServiceProviderServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.community.api.services.ServiceProvider;
+
+import com.community.api.endpoint.serviceProvider.ServiceProviderEntity;
+import com.community.api.endpoint.serviceProvider.ServiceProviderStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.social.ResourceNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import java.lang.reflect.Field;
+
+@Service
+public class ServiceProviderServiceImpl implements ServiceProviderService {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    @Transactional
+    public ServiceProviderEntity saveServiceProvider(ServiceProviderEntity serviceProviderEntity) {
+        try {
+            entityManager.persist(serviceProviderEntity);
+            ServiceProviderStatus serviceProviderStatus=entityManager.find(ServiceProviderStatus.class,1);
```

Comment:
create a function for finding status

like this and use here 
            ServiceProviderStatus serviceProviderStatus = findServiceProviderStatus(1);
            
            and create seprate function
            
            
                private ServiceProviderStatus findServiceProviderStatus(Integer id) {
        return entityManager.find(ServiceProviderStatus.class, id);
    }